src/client/app/shared/alert.class.ts

```diff
@@ -4,3 +4,8 @@ export class Alert {
 		public type: string,
 		public msg: string) {}
 }
+
+export class Alerts {
+	list: Alert[] = [];
+	map: Map<number, Alert> = new Map<number, Alert>();
```

Comment:
I know this wasn't in the original that you took this from, but seems like we'd want add and remove functions that add an alert to both the list and the map.  That can be separate PR though